extensions/core/deployment/src/main/java/org/apache/camel/quarkus/core/deployment/CamelSupport.java

```diff
@@ -56,4 +69,67 @@ public static boolean isPublic(ClassInfo ci) {
             .flatMap(CamelSupport::safeWalk)
             .filter(Files::isRegularFile);
     }
+
+    public static Stream<String> getRouteBuilderClasses(IndexView view) {
+        Set<ClassInfo> allKnownImplementors = new HashSet<>();
```

Comment:
I wonder if we may want to use a LinkedHashSet to have ordering of discovered classes

Reply:
The problem is that the order also depends on how Jandex indexes entries so I'm not sure if it matter, maybe better is to have the RouteBuilder extends the Ordered interface and order them when routes are processed ?

Reply:
Ah yeah this allows end users to control this if they have use-cases where this may matter. I dont recall if we in eg spring-boot discover of routes have any ordering too. But the Ordered is a good idea, we use it elsewhere in Camel too.

Reply:
As far as I remember you can order things in spring-boot but we can assume that a routes that have an explicit order should take the precedence over any other ordering mechanics.

Needs to be tested but we can also document that the preferred way of ordering routes is through the Camel's Ordered interface.

Reply:
https://issues.apache.org/jira/browse/CAMEL-14031